HotelApp/Models/Sort.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Dynamic.Core;
+using System.Reflection;
+using System.Text;
+
+namespace HotelApp.API.Models
+{
+    public class Sort<T> : ISort<T>
+    {
+        public IQueryable<T> ApplySort(IQueryable<T> entities, string orderBy)
+        {
+            if (!entities.Any())
+            {
+                return entities;
+            }
+            if (string.IsNullOrWhiteSpace(orderBy))
+            {
+                return entities;
+            }
+            var parameters = orderBy.Trim().Split(',');
+            var propertyInfos = typeof(T).GetProperties(BindingFlags.Public | BindingFlags.Instance);
+            var entityQueryBuilder = new StringBuilder();
+
+            foreach (var param in parameters)
+            {
+                if (string.IsNullOrWhiteSpace(param))
+                {
+                    continue;
+                }
+                var propertyFromQueryName = param.Split(" ")[0];
+                var objectProperty = propertyInfos.FirstOrDefault(pi =>
+                pi.Name.Equals(propertyFromQueryName,
+                               StringComparison.InvariantCultureIgnoreCase));
+
+                if (objectProperty == null)
+                {
+                    continue;
+                }
+                var sortingOrder = param.EndsWith(" desc") ? "descending" : "ascending";
+
+                entityQueryBuilder.Append($"{objectProperty.Name} {sortingOrder}, ");
+            }
+
+            var entityQuery = entityQueryBuilder.ToString().TrimEnd(',', ' ');
+
+            return entities.OrderBy(entityQuery);
```

Comment:
Za sorting bi ja radije išao ručno napisat if ili switch uvjete ovisno o nekim fiksnim parametrima. Ovo je super jer onda nemaš puno zezancije na svakom mogućem entitetu, ali me strah pošto je sortiranje akcija koja se često poziva, a reflection zna biti dosta sporiji od običnog koda. 
Istina da se tu radi o tipa mikro ili čak nano sekundama, ali na nekoj dosta većoj aplikaciji sa puno usera  i na nekom većem chunku koda i to isto znači nešto

Ostavi sad ovak, ali da znaš da treba razmišljat i o tim stvarima